### DISCUSSION TITLE
Not1mm and wfview: Invalid start byte on rigctld

### DISCUSSION THREAD

**Discussion by VE2ZAZ** [mbridak/not1mm#237] [General]
Hi,
Has anyone tried using Not1mm with wfview's rigcltd server? On my Kubuntu 24.04LTS machine, I get errors in Not1mm:
UnicodeDecodeError:`` 'utf-8' codec can't decode byte 0xfd in position 18: invalid start byte
Traceback (most recent call last):
  File "/home/zaz/.local/lib/python3.12/site-packages/not1mm/vfo.py", line 244, in poll_radio
    vfo = self.rig_control.get_vfo()
          ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zaz/.local/lib/python3.12/site-packages/not1mm/lib/cat_interface.py", line 256, in get_vfo
    vfo = self.__getvfo_rigctld()
          ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zaz/.local/lib/python3.12/site-packages/not1mm/lib/cat_interface.py", line 288, in __getvfo_rigctld
    report = self.__get_serial_string().strip()
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zaz/.local/lib/python3.12/site-packages/not1mm/lib/cat_interface.py", line 151, in __get_serial_string
    thegrab += self.rigctrlsocket.recv(1024).decode()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xfd in position 27: invalid start byte

The byte position reported varies from packet to packet, but otherwise, it is a constant flow of these errors, and nothing gets decoded from the radio.

When wfview is terminated, these errors stop showing. So there is some data flow, however not a single packet gets decoded properly, and the radio symbol stays red or gray.

My wfview session is connected to an IC-7300, and everything works normally between wfview and the radio.
wfview settings: rigctld enabled, port 4533, TCP Server port 50005.
Not1mm settings: IP 127.0.0.1, port 4533 or port 50005, rigctld selected.

Thanks in advance for your help, 73,

Bert, VE2ZAZ/VA2IW
https://github.com/VE2ZAZ?tab=repositories
https://ve2zaz.net

**Comment by mbridak** [comment]
Hmm, I have not. Curious, from a terminal could you try:

`telnet 127.0.0.1 4533`

then press `f` then enter then `m` then enter and see if you get anything like below?

```bash
 mbridak@fredo  ~  telnet 127.0.0.1 4532
Trying 127.0.0.1...
Connected to 127.0.0.1.
Escape character is '^]'.
f
14250000
m
USB
2400
```

**Reply by VE2ZAZ** [reply]
Thanks Michael for following up. This is what I get:

~$ telnet 127.0.0.1 4533
Trying 127.0.0.1...
Connected to 127.0.0.1.
Escape character is '^]'.
f
28034600
m
CW
300

Looks like I can manually talk to the rigctld server. BTW, that server works fine with n1mm+.

Thanks, 73,

Bert.

**Comment by mbridak** [comment]
Hmm. 
That's perplexing.
I'll try and carve out some time this weekend and install wfview in the shack and see if I can replicate this.

Cheers, Mike K6GTE
